lineax/_solver/cholesky.py

```diff
@@ -77,8 +77,14 @@ def compute(
         return solution, RESULTS.successful, {}
 
     def transpose(self, state: _CholeskyState, options: dict[str, Any]):
-        # Matrix is symmetric anyway
-        return state, options
+        # Matrix is self-adjoint
+        factor, is_nsd = state
+        return (factor.conj(), is_nsd), options
```

Comment:
This is a good catch!
This isn't the only solver assuming the the matrix is PSD/symmetric/etc -- I think some of the others may also need updating?
(This is kind of finickity isn't it?)

Reply:
Definitely possible that the tests need enhancing to catch this -- after all we don't really have any complex tests at all, since that's not really a feature yet.
The fact that we only ever do `conj(transpose(...))` as a pair might also cover some mistakes if the pair is correct even when each individual one is wrong.

I think it'd worth adding some complex tests that explicitly check calling these methods individually.

Reply:
Hmm. At the very least I'd expect the CG tests to fail given complex input: this also requires the operator to be positive or negative definite. I think this indicates we need a lot more tests for complex input!

(And, side note, CG currently appears to have a pointless no-op transpose in the `normal=False` case -- although that's an existing thing, not something changed in this PR.)

Reply:
I've patched psd matrix generation earlier:
https://github.com/google/lineax/blob/199c27777dea454f6a7d23c531664dba9bd2cbf9/tests/helpers.py#L54-L57
so I don't thing CG should fail.

The problem here was in implicit assumption psd->symmetric

Reply:
By briefly checking other places mentioning symmetric operators I think only this one is also problematic:
https://github.com/google/lineax/blob/199c27777dea454f6a7d23c531664dba9bd2cbf9/lineax/_operator.py#L1407-L1416

Reply:
Specifically all operator/solver transpositions look ok to me